src/BluetoothAdapter.cpp

```diff
@@ -139,6 +139,7 @@ std::unique_ptr<BluetoothAdapter> BluetoothAdapter::make(Object *object,
         (adapter = object_get_adapter1(object)) != NULL) {
 
         std::unique_ptr<BluetoothAdapter> p(new BluetoothAdapter(adapter));
+        g_object_unref(adapter);
```

Comment:
Please don't do double dereferencing like this. Bluetooth* classes will do a single reference in the constructor and dereference in the destructor. Doing a double dereference here will just destroy `adapter` before it's supposed to be destroyed. This goes for all functions in which you call g_object_unref twice. This will definitely cause issues. It's possible that the object gets leaked and there might be a bug causing that, but this is not the correct way to treat this bug.